File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        storeNum = {}

        for i , num in enumerate(nums):
            complement = target - num

            if complement in storeNum:
                return [storeNum[complement], i ]
            
            storeNum[num] = i
            
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       Map<Integer,Integer> map = new HashMap<>();
       for(int i =0 ; i < nums.length; i++){
        int x= target - nums[i];
        if(map.containsKey(x)){
            return new int[]{map.get(x),i};
        }
        map.put(nums[i],i);
       }
        return new int []{};
    }
}